schemaregistry/schemaregistry_client.go

```diff
@@ -265,14 +265,21 @@ func (c *client) Register(subject string, schema SchemaInfo, normalize bool) (id
 	metadata := SchemaMetadata{
 		SchemaInfo: schema,
 	}
-	err = c.restService.handleRequest(newRequest("POST", versionNormalize, &metadata, url.PathEscape(subject), normalize), &metadata)
-	if err != nil {
-		return -1, err
-	}
 	c.schemaCacheLock.Lock()
-	c.schemaCache.Put(cacheKey, metadata.ID)
+	// another goroutine could have already put it in cache
+	idValue, ok = c.schemaCache.Get(cacheKey)
+	if !ok {
+		err = c.restService.handleRequest(newRequest("POST", versionNormalize, &metadata, url.PathEscape(subject), normalize), &metadata)
```

Comment:
This will block all other threads/go-routines to access the schema cache during the duration of the REST call, which could take a long time. Is this intentional?

One way to avoid this is to also have a lock on each cache entry and only keep that lock held during the request, this way accesses to other schemas are not affected by retrieving this one. (Do pay attention to lock ordering though).